changelog/23366.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+auth/aws: Fixes a panic that can occur when an IAM-based login is attempted after config deletion.
+```
```

Comment:
Wonder if we should also add a `change`? Something like

```
release-note:change
auth/aws: Behavior change to no longer allow IAM-based login when the client configuration does not exist
```

Reply:
Changed the approach here to maintain the existing behavior. So I won't be including a change CL entry. Thanks!